### PR TITLE
Puppet 5 Modules for Enterprise Integrator 6.4.0

### PR DESCRIPTION
**Pull request by irunika**

### CODE REVIEW

File: modules/ei_analytics_dashboard/templates/carbon-home/wso2/analytics/conf/dashboard/deployment.yaml.erb [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+        # Data receiver type
+        # THIS IS A MANDATORY FIELD
+      type: Binary
+        # Data receiver properties
+      properties:
+        tcpPort: '9611'
+        sslPort: '9711'
+        tcpReceiverThreadPoolSize: '100'
+        sslReceiverThreadPoolSize: '100'
+        hostName: 0.0.0.0
```
Comment (VimukthiPerera) [inline modules/ei_analytics_dashboard/templates/carbon-home/wso2/analytics/conf/dashboard/deployment.yaml.erb]: Shall we parameterize hostName here?

File: modules/ei_analytics_dashboard/templates/carbon-home/wso2/analytics/conf/dashboard/deployment.yaml.erb [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+        # Data agent name
+        # THIS IS A MANDATORY FIELD
+      name: Thrift
+        # Data endpoint class
+        # THIS IS A MANDATORY FIELD
+      dataEndpointClass: org.wso2.carbon.databridge.agent.endpoint.thrift.ThriftDataEndpoint
+        # Data publisher strategy
+      publishingStrategy: async
+        # Trust store path
+      trustStorePath: '${sys:carbon.home}/resources/security/client-truststore.jks'
```
Comment (VimukthiPerera) [inline modules/ei_analytics_dashboard/templates/carbon-home/wso2/analytics/conf/dashboard/deployment.yaml.erb]: Trust store should be parameterized

File: modules/ei_analytics_worker/manifests/init.pp [hunk 3]
```diff
@@ -107,7 +59,7 @@
     source => "puppet:///modules/${module_name}/${ei_package}",
   }
 
-  # Install WSO2 Enterprise Integrator
+  # Install WSO2 API Manager
```
Comment (VimukthiPerera) [inline modules/ei_analytics_worker/manifests/init.pp]: This should be Enterprise Integrator

File: modules/ei_analytics_worker/manifests/init.pp [hunk 4]
```diff
@@ -155,7 +107,7 @@
   /*
     Following script can be used to copy file to a given location.
     This will copy some_file to install_path -> repository.
-    Note: Ensure that file is available in modules -> ei_analytics -> files
+    Note: Ensure that file is available in modules -> apim -> files
```
Comment (VimukthiPerera) [inline modules/ei_analytics_worker/manifests/init.pp]: Here, apim should be ei_analytics_worker.

File: modules/ei_bps/templates/carbon-home/wso2/business-process/conf/carbon.xml.erb [hunk 5]
```diff
@@ -382,6 +408,23 @@
             <KeyPassword><%= @security_keystore_key_password %></KeyPassword>
         </KeyStore>
 
+        <!--
+            The KeyStore which is used for encrypting/decrypting internal data.
+            This block is read by Carbon Crypto Service.
+        -->
+        <InternalKeyStore>
+            <!-- Keystore file location-->
+            <Location>${carbon.home}/repository/resources/security/wso2carbon.jks</Location>
```
Comment (VimukthiPerera) [inline modules/ei_bps/templates/carbon-home/wso2/business-process/conf/carbon.xml.erb]: Shall we parameterize internal key store as well?

File: modules/ei_integrator/templates/carbon-home/conf/carbon.xml.erb [hunk 6]
```diff
@@ -374,6 +400,23 @@
             <KeyPassword><%= @security_keystore_key_password %></KeyPassword>
         </KeyStore>
 
+        <!--
+            The KeyStore which is used for encrypting/decrypting internal data.
+            This block is read by Carbon Crypto Service.
+        -->
+        <InternalKeyStore>
+            <!-- Keystore file location-->
+            <Location>${carbon.home}/repository/resources/security/wso2carbon.jks</Location>
```
Comment (VimukthiPerera) [inline modules/ei_integrator/templates/carbon-home/conf/carbon.xml.erb]: The internal key store should be parametrized.

File: modules/ei_micro_integrator/templates/carbon-home/wso2/micro-integrator/conf/carbon.xml.erb [hunk 7]
```diff
@@ -374,6 +400,23 @@
             <KeyPassword><%= @security_keystore_key_password %></KeyPassword>
         </KeyStore>
 
+        <!--
+            The KeyStore which is used for encrypting/decrypting internal data.
+            This block is read by Carbon Crypto Service.
+        -->
+        <InternalKeyStore>
+            <!-- Keystore file location-->
+            <Location>${carbon.home}/repository/resources/security/wso2carbon.jks</Location>
```
Comment (VimukthiPerera) [inline modules/ei_micro_integrator/templates/carbon-home/wso2/micro-integrator/conf/carbon.xml.erb]: There are several places where the internal key store is not parameterized.

File: modules/ei_analytics_dashboard/templates/carbon-home/wso2/analytics/conf/dashboard/deployment.yaml.erb [hunk 8]
```diff
@@ -0,0 +1,466 @@
...
+#        validationTimeout: 30000
+#        isAutoCommit: false
+  - name: Message_Tracing_DB
+    description: "The datasource used for message tracer to store span information."
+    jndiConfig:
+      name: jdbc/Message_Tracing_DB
+    definition:
+      type: RDBMS
+      configuration:
+        jdbcUrl: 'jdbc:h2:${sys:carbon.home}/wso2/dashboard/database/MESSAGE_TRACING_DB;AUTO_SERVER=TRUE'
```
Comment (VimukthiPerera) [inline modules/ei_analytics_dashboard/templates/carbon-home/wso2/analytics/conf/dashboard/deployment.yaml.erb]: Shall we parameterize this as well? For the worker, this has already being done.

File: README.md [hunk 9]
```diff
@@ -129,5 +129,4 @@ This is used to add custom user code to the profile.
 - params.pp <br>
 This contains all the necessary parameters for main configurations and template rendering.
 - startserver.pp <br>
-This runs finally and starts the server as a service.
-=======
+This runs finally and starts the server as a service
```
Comment (VimukthiPerera) [inline README.md]: Shall we end the sentence with a full-stop?

**Review by VimukthiPerera — APPROVED** [review]

**Review by DilanUA — APPROVED** [review]

**Pull request merged**